src/main/java/net/dries007/tfc/objects/blocks/agriculture/BlockCropSimple.java

```diff
@@ -101,52 +64,47 @@ public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state,
         return false;
     }
 
-    @Override
-    public void harvestBlock(World worldIn, EntityPlayer player, BlockPos pos, IBlockState state, @Nullable TileEntity te, ItemStack stack)
+    public static BlockCropSimple create(ICrop crop, boolean isPickable)
     {
-        // todo: in 1.14 change to use the apply skill loot table
-        super.harvestBlock(worldIn, player, pos, state, te, stack);
-
-        ItemStack seedStack = new ItemStack(ItemSeedsTFC.get(crop));
-        ItemStack foodStack = crop.getFoodDrop(state.getValue(getStageProperty()));
-        SimpleSkill skill = CapabilityPlayerData.getSkill(player, SkillType.AGRICULTURE);
-        if (skill != null)
-        {
-            foodStack.setCount(1 + RANDOM.nextInt(2 + (int) (6 * skill.getTotalLevel())));
-            if (skill.getTier().isAtLeast(SkillTier.ADEPT) && RANDOM.nextInt(10 - 2 * skill.getTier().ordinal()) == 0)
-            {
-                seedStack.setCount(2);
-            }
-            skill.add(0.04f);
-        }
-        if (!seedStack.isEmpty())
-        {
-            InventoryHelper.spawnItemStack(worldIn, pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5, seedStack);
-        }
-        if (!foodStack.isEmpty())
+        switch (crop.getMaxStage() + 1)
         {
-            InventoryHelper.spawnItemStack(worldIn, pos.getX() + 0.5, pos.getY() + 0.5, pos.getZ() + 0.5, foodStack);
+            case 5:
+                return new BlockCropSimple(crop, isPickable)
+                {
+                    @Override
+                    public PropertyInteger getStageProperty()
+                    {
+                        return STAGE_5;
+                    }
+                };
+            case 6:
+                return new BlockCropSimple(crop, isPickable)
+                {
+                    @Override
+                    public PropertyInteger getStageProperty()
+                    {
+                        return STAGE_6;
+                    }
+                };
+            case 7:
+                return new BlockCropSimple(crop, isPickable)
+                {
+                    @Override
+                    public PropertyInteger getStageProperty()
+                    {
+                        return STAGE_7;
+                    }
+                };
+            case 8:
+                return new BlockCropSimple(crop, isPickable)
+                {
+                    @Override
+                    public PropertyInteger getStageProperty()
+                    {
+                        return STAGE_8;
+                    }
+                };
```

Comment:
Instead of a switch over the property for each different crop type (imo ugly, long, duplicated code), consider seperating these into two methods:
```java
public static void getStagePropertyForValue(int maxStages) {
    // return the stage property based on maxStages, perhaps even using an array index
}
public static BlockCropSimple create(int maxStages) {
    PropertyInteger property = getStagePropertyForValue(maxStages);
    return new BlockCropSimple(...) {
        @Override public PropertyInteger getStageProperty() { return property; }
    }
}
```

Reply:
I've removed the switch and replaced it with a method like you've suggested that uses the stage mapping.